### PR TITLE
bugfix: Uniswap constructor and "check_approval" decorator throwing whenever passing kwargs. Issue #218

### PR DESCRIPTION
**Pull request by RoscoeTheDog**
Asked the chatbot to come up with a solution. Seems to work fine. I'm not too familiar with all the internal relationships, so someone should review before merging. 👍

### CODE REVIEW

**Comment by RoscoeTheDog** [comment]
Also added *args, **kwargs to Uniswap constructor

When creating a wrapper class for this API, I encountered issues passing down a flattened dict of kwargs to the constructor of Uniswap() class due to it not having *args, **kwargs in it's constructors signature. Any extra arguments that were not explicitily in the constructors signature would throw an error, which makes it less friendly for other software systems to wrap it's code with.

Simply adding *args and **kwargs to the constructor fixes this issue and doesn't intrtoduce any negative effects.

File: uniswap/uniswap.py [hunk 1]
```diff
@@ -92,6 +92,8 @@ def __init__(
         factory_contract_addr: Optional[str] = None,
         router_contract_addr: Optional[str] = None,
         enable_caching: bool = False,
+		*args,
+		**kwargs
```
Comment (ErikBjare) [inline uniswap/uniswap.py:96]: Unused. Don't do this, clean up your args instead.

File: uniswap/decorators.py [hunk 2]
```diff
@@ -20,28 +21,30 @@
...
-    method: Callable[Concatenate["Uniswap", P], T]
-) -> Callable[Concatenate["Uniswap", P], T]:
-    """Decorator to check if user is approved for a token. It approves them if they
+def check_approval(method: Callable[..., T]) -> Callable[..., T]:
+    """Decorator to check if the user is approved for a token. It approves them if they
     need to be approved."""
 
     @functools.wraps(method)
-    def approved(self: "Uniswap", *args: P.args, **kwargs: P.kwargs) -> T:
+    def approved(self: "Uniswap", *args: Union[AddressLike, int], **kwargs: AddressLike) -> T:
```
Comment (ErikBjare) [inline uniswap/decorators.py:29]: This ruins typing (and is not even valid, `*args` has to be an iterable and `**kwargs` a mapping) for any function annotated with `check_approval`. The `ParamSpec` is used for a reason.

**Review by ErikBjare — COMMENTED** [review]
>Any extra arguments that were not explicitily in the constructors signature would throw an error

That is intended. Don't pass things that aren't going to be used.

>which makes it less friendly for other software systems to wrap it's code with.

No, it just makes you have to clean up your args (as you should anyway).

>Simply adding *args and **kwargs to the constructor fixes this issue and doesn't intrtoduce any negative effects.

It's not a bug, it's a feature. Adding unused args and kwargs *does* introduce negative effects. It ruins typecheckers ability to spot misspelled keyword arguments and otherwise unused args, among other things.

As explained above, this PR breaks typing in several places (see CI), so it will not get merged.

**Comment by ErikBjare** [comment]
If you want a fix for https://github.com/uniswap-python/uniswap-python/issues/218, feel free to submit a PR that checks the `kwargs` for the token in addition to the `args`.

**Pull request closed without merging**